### PR TITLE
Update week 1

### PR DESCRIPTION
**Pull request by AliSchlereth**
@neight-allen @case-eee 

I added some language around Abstraction and Encapsulation to some M1 week 1 lessons. My aim here was to weave in a *very* brief introduction to these concepts while we are teaching other things.  

Is the information I've included correct? Do the places I've connected these concepts into our learning make sense?  
Thanks!

### CODE REVIEW

File: module1/lessons/intro_to_classes.md [hunk 1]
```diff
@@ -233,7 +263,7 @@ end
 
 
 
-Changing our `attr_reader`s to `attr_accessor`s makes it so that we can change the classroom's value from outside the class to **anything** we want. We are placing some trust in people using the class that they will use it responsibly.
+Changing our `attr_reader`s to `attr_accessor`s makes it so that we can change the classroom's value from outside the class to **anything** we want. Note, we are changing this object's encapsulation. We are placing some trust in people using the class that they will use it responsibly.
```
Comment (neight-allen) [inline module1/lessons/intro_to_classes.md]: Changing and even _reducing_ encapsulation

File: module1/lessons/intro_to_methods.md [hunk 2]
```diff
@@ -168,7 +174,7 @@ Once you've completed that, see if you can determine what the `print_converted`
 
 ### Layers of Abstraction
 
-One of the advantages of using methods is that we can build methods that operate at higher levels of abstraction than other methods. In some ways, this is like a pyramid where higher level methods rely on lower level methods to take care of the details.
+One of the advantages of using methods is that we can build methods that operate at higher levels of abstraction than other methods. Abstraction is a practice where less complex functionality is exposed in an interface and more complex funtionality is suppressed. In some ways, this is like a pyramid where higher level methods rely on lower level methods to take care of the details.
```
Comment (neight-allen) [inline module1/lessons/intro_to_methods.md]: Oh, this is where the abstract-o-meter goes.

File: module1/lessons/intro_to_oop.md [hunk 3]
```diff
@@ -324,3 +342,4 @@ Feeling fancy? Update your Snack class to have a volume. Update the `add_snack`
 * What are some examples of classes and instances from everyday life?
 * How would you define a `Person` class that had height, weight, and age attributes in Ruby?
 * What are the two main responsibilities of an instance of an object in Ruby?
+* How did we apply Abstraction and Encapsulation today?
```
Comment (neight-allen) [inline module1/lessons/intro_to_oop.md]: I would love to have a question like "Explain Abstraction in your own words" before they talk about how they used it.

Comment (AliSchlereth) [inline module1/lessons/intro_to_oop.md]: I agree.  The order of these lessons is Intro to Methods, Intro to Classes, Intro to OOP. The first definition comes in Intro to Methods, but I actually think it is easier to think of Abstraction on the class level than method level. Maybe we put a Remember Abstraction means... in addition to pointing out that objects are an example of Abstraction into the Intro to Classes lesson and then put that question in the WrapUp section. Does that sound good?

File: module1/lessons/intro_to_classes.md [hunk 4]
```diff
@@ -75,7 +89,9 @@ First we get our expected output from running the `.convert` method on our `Conv
 
 What's after that?
 
-The output from `puts "This is our converter: #{converter}"` is the way that Ruby represents an instance of an object to use in print. Notice there's an `object_id` that is unique to each instance we've created. Even though they generally look the same, our computer is now tracking three separate instances of our Converter object.
+The output from `puts "This is our converter: #{converter}"` is the way that Ruby represents an instance of an object to use in print. Notice there's an `object_id` that is unique to each instance we've created. Even though they generally look the same, our computer is now tracking three separate instances of our Converter object.  
+
+An *object* is an abstract representation of a real world thing. Here we see Abstraction coming in to play where the user can interface with a Converter. It has specific details about it that we interact with such as `.convert` but other details that we do not interact with such as .`convert_to_celcius`.
```
Comment (AliSchlereth) [inline module1/lessons/intro_to_classes.md]: Yes! I love this. I'll put it in the Method lesson.

File: module1/lessons/intro_to_methods.md (hunk 2 above)
Comment (AliSchlereth) [inline module1/lessons/intro_to_methods.md]: Looking back on this now, this explanation of abstraction sounds a lot like encapsulation. I understand how/why it is not, but will we confuse them more by using words like expose and suppress? Any suggestions on how to phrase it differently when it comes to a very basic explanation relating to methods?

File: module1/lessons/intro_to_classes.md [hunk 5]
```diff
@@ -8,7 +8,22 @@ tags: ruby, classes, objects
...
-* Define classes with instance variables in Ruby
+* Define classes with instance variables in Ruby  
+
+## Schedule 
+* 5min - WarmUp
+* 20min - Organizign Methods with Classes 
+* 
+* 5min - WrapUp
+
+## Vocabulary
```
Comment (case-eee) [inline module1/lessons/intro_to_classes.md]: 👍

File: module1/lessons/intro_to_classes.md [hunk 6]
```diff
@@ -112,7 +128,20 @@ puts classroom_a
 
 Run that and what do we get? Sure enough, it's an instance of classroom!
 
-That's great. We can set the attributes of the classroom instance when I create it. What if I want to access those attributes? Basically, what happens if I forget the length of the classroom? How do I ask it?
+That's great. We can set the attributes of the classroom instance when we create it. What if I want to access those attributes? Basically, what happens if I forget the length of the classroom? How do I ask it?
+We could put a pry to dig around. 
+
+```ruby 
+#classroom_runner.rb
```
Comment (case-eee) [inline module1/lessons/intro_to_classes.md]: love this!

File: module1/lessons/intro_to_classes.md [hunk 7]
```diff
@@ -112,7 +128,20 @@ puts classroom_a
...
+require 'pry'
+require './classroom'
+
+classroom_a = Classroom.new(10, 5, 20)
+binding.pry
+```
+
+When we call `classroom_a` in our pry session, we get something like this `#<Classroom:0x007fe119e926c0>`, we still can't see the length. This is due in part to the principle of Encapsulation, where information is only exposed when intentionally built to do so.
```
Comment (case-eee) [inline module1/lessons/intro_to_classes.md]: great 🎉

File: module1/lessons/intro_to_methods.md [hunk 8]
```diff
@@ -13,6 +13,12 @@ tags: ruby, methods
 * Define methods that do not take arguments
 * Define a class with instance methods
 
+## Vocabulary
```
Comment (case-eee) [inline module1/lessons/intro_to_methods.md]: again - 👍

File: module1/lessons/intro_to_methods.md [hunk 9]
```diff
@@ -168,7 +174,7 @@ Once you've completed that, see if you can determine what the `print_converted`
 
 ### Layers of Abstraction
 
-One of the advantages of using methods is that we can build methods that operate at higher levels of abstraction than other methods. In some ways, this is like a pyramid where higher level methods rely on lower level methods to take care of the details.
+One of the advantages of using methods is that we can build methods that operate at higher levels of abstraction than other methods. Abstraction is a practice where less complex functionality is exposed in an interface and more complex funtionality is suppressed. In some ways, this is like a pyramid where higher level methods rely on lower level methods to take care of the details.
```
Comment (case-eee) [inline module1/lessons/intro_to_methods.md]: functionality is spelled incorrectly here!

File: module1/lessons/intro_to_oop.md [hunk 10]
```diff
@@ -23,7 +29,7 @@ Available [here](../slides/intro_to_oop)
 
 #### Using Related Methods
 
-Let's start with our `Classroom` class from yesterday.
+Let's start with our `Classroom` class from yesterday. Remember a class is an abstract representation of a real world thing/concept.
```
Comment (case-eee) [inline module1/lessons/intro_to_oop.md]: i really like that you're throwing in the vocab here and throughout these lessons

**Pull request merged**